### PR TITLE
#166789864: Setup unit testing

### PR DESCRIPTION
**Pull request by Emile-Nsengimana**
#### What does this PR do?
setup unit testing and ensures that some functionality are tested
#### Description of Task to be completed?
Setup unit testing. Ensure all implemented functionality is tested henceforth.
#### How should this be manually tested?
- First clone it to your machine:
```
https://github.com/andela/ah-lobos-backend.git
```
- Open it using your favorite IDE, I used ([vs code](https://code.visualstudio.com/download))

- Install all necessary node modules
```
npm install
```
- Run test using
```
npm test
```
#### Any background context you want to provide?
- Create `.env` file with the help of `example.env`
- The project structure has been changed
#### What are the relevant pivotal tracker stories?
[#166789864](https://www.pivotaltracker.com/n/projects/2356563/stories/166789864)
#### Screenshots
N/A
#### Questions:
N/A

### CODE REVIEW

**Review by adafia — COMMENTED** [review]
Please resolve the blank line issue in your `.gitignore` file.

File: example.env [hunk 1]
```diff
@@ -1,2 +1,3 @@
-DATABASE_USERNAME_DEV="your-database-username"
-SECRET_KEY="your-database-password"
+DATABASE_USERNAME_DEV="username for development database"
+DATABASE_USERNAME_TEST="username for test database"
+DATABASE_PASSWORD="database password"
```
Comment (Dom58) [inline example.env]: Leave a new branch line

**Review by Dom58 — COMMENTED** [review]
@Emile-Nsengimana to what I know, it is not good to show a project structure in public ( like how you show that in the PR description ). To me you may edit and remove that project structure screenshot.

**Review by j4l13n — APPROVED** [review]

**Review by j4l13n — APPROVED** [review]

**Review by adafia — APPROVED** [review]

**Review by Dom58 — APPROVED** [review]

**Review by manzif — APPROVED** [review]

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+src/db/
```
Comment (emukungu) [inline .eslintignore]: new line missing here. 
- research about the editorconfig file.

File: package.json [hunk 3]
```diff
@@ -5,53 +5,51 @@
...
-    "test": "mocha  --require babel-register tests/*.js --exit",
+    "test": "npm run create_tables && NODE_ENV=test nyc --reporter=text mocha --require babel-register --require babel-polyfill ./tests/ --exit && npm run drop_tables",
     "coveralls": "npm test && nyc report --reporter=text-lcov | coveralls",
     "coverage": "npm test && nyc report --reporter=text-lcov | coveralls",
-    "start": "babel-node index.js",
-    "dev": "nodemon --exec babel-node index.js"
+    "drop_tables": "NODE_ENV=test sequelize db:migrate:undo:all",
+    "create_tables": "NODE_ENV=test sequelize db:migrate",
+    "start": "node --exec babel-node ./src/index.js",
+    "dev": "npm run create_tables && nodemon --exec babel-node ./src/index.js"
```
Comment (emukungu) [inline package.json]: you can't create tables every time you make a change to the code. They should be created once.
Remove create tables in the dev command

The coverage and coveralls commands are the same. Remove one of them.

**Comment by emukungu** [comment]
Edit your commit message to a better message in the body.

Coverage was at 100% before, why the drop?

**Comment by emukungu** [comment]
gitignore the editorconfig file. It's for your IDE to follow the conventions, don't need it as part of the code base.

**Pull request merged**